static/app/views/aiMonitoring/aiMonitoringDetailsPage.tsx

```diff
@@ -25,30 +23,32 @@ import {
 import {PipelineSpansTable} from 'sentry/views/aiMonitoring/pipelineSpansTable';
 import {MetricReadout} from 'sentry/views/performance/metricReadout';
 import * as ModuleLayout from 'sentry/views/performance/moduleLayout';
+import {ModulePageProviders} from 'sentry/views/performance/modulePageProviders';
 import {useSpanMetrics} from 'sentry/views/starfish/queries/useDiscover';
 import {
   SpanFunction,
   SpanMetricsField,
   type SpanMetricsQueryFilters,
 } from 'sentry/views/starfish/types';
 
-function NoAccessComponent() {
-  return (
-    <Layout.Page withPadding>
-      <Alert type="warning">{t("You don't have access to this feature")}</Alert>
-    </Layout.Page>
-  );
-}
 interface Props {
   params: {
     groupId: string;
   };
 }
 
-export default function AiMonitoringPage({params}: Props) {
+type Query = {
+  'span.description'?: string;
+};
+
+export function AiMonitoringPage({params}: Props) {
+  const location = useLocation<Query>();
+
   const organization = useOrganization();
   const {groupId} = params;
 
+  const spanDescription = decodeScalar(location.query?.['span.description']);
```

Comment:
smells like XSS but I'll let it pass.

Reply:
This is a good comment, so I don't want to leave it unanswered before merging. The "smell" here is that I'm rendering URL content into the DOM. This does _smell_ like a reflected XSS vulnerability, but it's safe.

The attack vector is crafting a URL like `https://<org>.sentry.io/ai-monitoring/pipeline-type/<HASH>/?span.description=<script>alert("you win")</script>`. The reason this is _not_ an XSS vulnerability is that React safely escapes all rendered content:

<img width="434" alt="Screenshot 2024-05-14 at 10 07 08 AM" src="https://github.com/getsentry/sentry/assets/989898/2aa9de92-da89-40ab-8af5-cbb422521da0">

In contrast, using `dangerouslySetInnerHTML` _would_ create a reflected XSS attack here.